### PR TITLE
React to updated message of guest user

### PR DESCRIPTION
**Pull request by Fecony**
Listen for `messageUpdated` event, if it was edited by guest user and is more than 20 chars then remove guest role

> Needs testing because locally my bot doesn't have permissions to manage roles and idk why I tried to give him everything 🥲

### CODE REVIEW

File: src/discord-bot.js [hunk 1]
```diff
@@ -89,6 +89,27 @@ const start = () => {
...
     }
   })
+
+  bot.on('messageUpdate', async (oldMessage, newMessage) => {
+    const member = newMessage.guild.member(newMessage.author);
+
+    if (
+      newMessage.channel.id.toString() === INTRODUCTIONS_CHANNEL_ID &&
+      member.roles.cache.get(GUEST_ROLE_ID) &&
+      oldMessage.content.length < 20
```
Comment (Fecony) [inline src/discord-bot.js:99]: It may not be necessary because if old message was <20 chars, user still has guest role

**Review by Martinsos — CHANGES_REQUESTED** [review]
@Fecony thanks for this, it looks good!

Did you find a way to test this, I guess not, it is pretty tricky to do it? If not I can test it once we are done.

Ok, so basically we want to run the same check if new message arrives, or if old message was updated, and it happened in introductions channel, and user has GUEST role. That makes sense to me.

I think this is good but we should remove duplication. We want one piecce of logic to handle both situations. Should we make a function? Smth like `function ifGuestIntroductionMessageHandleIt (newMessage, oldMessage = undefined)`. And then we would just call it at those two places, and almost all of the logic would be in it?

**Comment by Fecony** [comment]
@Martinsos 
> Did you find a way to test this, I guess not, it is pretty tricky to do it? If not I can test it once we are done.

I have test bot where I tried to launch node + bot in demo server and it works. BUT I cannot add it a permission to remove roles and it fails on that part of code for me

> I think this is good but we should remove duplication. We want one piecce of logic to handle both situations. Should we 

Removed code duplication for `message` and `messageUpdated` looks like it really is just a same handler function if you don't want to change message for update or something
And `oldMessage` is not used really...

File: src/discord-bot.js [hunk 2]
```diff
@@ -55,43 +55,47 @@ const start = () => {
...
-      } else if (msg.content.includes('monthly')) {
-        await sendAnalyticsReport(bot, 'monthly')
-      } else if (msg.content.includes('daily')) {
-        await sendAnalyticsReport(bot, 'daily')
-      } else if (msg.content.includes('total'))  {
-        await sendAnalyticsReport(bot, 'total')
-      } else {
-        await sendAnalyticsHelp(bot)
-      }
+  if (msg.content.startsWith('!analytics') && msg.channel.id.toString() === REPORTS_CHANNEL_ID) {
```
Comment (Martinsos) [inline src/discord-bot.js:86]: I thought we would handle only introductions messages both on new message and upatedMessage, not that we will also handle !analytics bot commands.

The way you did it now, we allow people to update their analytics commands. Well, maybe that is ok at the end? I guess it is not a bit problem! Ok, let's leave it like this then, sounds good actually.

File: src/discord-bot.js [hunk 3]
```diff
@@ -55,43 +55,47 @@ const start = () => {
...
+    return;
+  }
+
+  const member = msg.guild.member(msg.author)
+
+  if (msg.channel.id.toString() === INTRODUCTIONS_CHANNEL_ID && member.roles.cache.get(GUEST_ROLE_ID)) {
+    const trimmedMsg = msg.content.trim().length;
+    if (trimmedMsg < 20) {
+      return msg.reply(
+        `\n👋 Great to have you here! Pls introduce yourself with a message that's at least 2️⃣0️⃣ characters long and I will give you full access to the server. \n❌ BTW editing your old message won't work b/c I'm not a very smart bot🤖. Doh!`
```
Comment (Martinsos) [inline src/discord-bot.js:75]: Oh we have to update now this line since editing old messages works now!

Comment (Fecony) [reply src/discord-bot.js]: @Martinsos Updated by removing the self-hatred message :boi:

File: src/discord-bot.js (hunk 2 above)
Comment (Fecony) [reply src/discord-bot.js:86]: Then I will leave it as is?
It probably makes sense to leave it just in `on:message` but I don't think it changes something

**Comment by Martinsos** [comment]
Looking good, thanks @Fecony , I am merging it!!!

**Review by Martinsos — APPROVED** [review]
LGTM!

**Pull request merged**